src/Exchange.ts

```diff
@@ -0,0 +1,435 @@
+import { Event } from './Event'
+import { Utils } from './Utils'
+import { User } from './User'
+import { Transaction } from './Transaction'
+import { CurrencyNetwork } from './CurrencyNetwork'
+import { Payment } from './Payment'
+import { ExchangeOptions, OrderbookOptions, Order } from './typings'
+
+import { BigNumber } from 'bignumber.js'
+import { BN } from 'bn.js'
+import * as ethUtils from 'ethereumjs-util'
+import * as ethABI from 'ethereumjs-abi'
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+export class Exchange {
+
+  constructor (
+    private event: Event,
+    private user: User,
+    private utils: Utils,
+    private transaction: Transaction,
+    private currencyNetwork: CurrencyNetwork,
+    private payment: Payment
+  ) {}
+
+  public getExchanges (): Promise<any> {
+    return this.utils.fetchUrl('exchange/exchanges')
+  }
+
+  public getEthWrappers (): Promise<any> {
+    return this.utils.fetchUrl('exchange/eth')
+  }
+
+  public async getOrderbook (
+    baseTokenAddress: string,
+    quoteTokenAddress: string,
+    { baseTokenDecimals, quoteTokenDecimals }: OrderbookOptions = {}
+  ): Promise<any> {
```

Comment:
Can we here maybe use the actual return type?

Reply:
done